lib/Analysis/Utility.cpp

```diff
@@ -194,16 +192,54 @@ bool ScanLoweringHelper::isSupported() {
 
 unsigned ScanLoweringHelper::getScratchSizeInBytes() {
   auto type = scanOp.getOperand(0).getType().cast<RankedTensorType>();
-  unsigned numElement =
+  unsigned tensorSizeInBytes =
       type.getNumElements() * type.getElementTypeBitWidth() / 8;
-  return numElement /
-         (getAxisNumElementsPerThreads() * getAxisNumThreadsPerWarp());
+  return tensorSizeInBytes /
+         (getAxisNumElementsPerThread() * getAxisNumThreadsPerWarp());
 }
 
 triton::gpu::BlockedEncodingAttr ScanLoweringHelper::getEncoding() {
   return srcEncoding.cast<triton::gpu::BlockedEncodingAttr>();
 }
 
+unsigned ScanLoweringHelper::getAxisElementStride() {
+  auto order = triton::gpu::getOrder(srcEncoding);
+  unsigned stride = 1;
+  for (unsigned dim : order) {
+    if (dim == getAxis())
+      return stride;
+    stride *= getEncoding().getSizePerThread()[dim];
```

Comment:
This can be replaced by `getContigPerThread(srcEncoding)[dim]`, which is a little bit more general